p2p/discover/udp.go

```diff
@@ -504,6 +504,28 @@ func (t *udp) send(toaddr *net.UDPAddr, ptype byte, req interface{}) error {
 	if err != nil {
 		return err
 	}
+	switch ptype {
```

Comment:
Same as above. The problem is that ping/pong packets are extremely frequent so there might be a performance penalty.

Reply:
So because this is the same strategy as `glog.V`, here's what the docs says:

> The second form is shorter but the first is cheaper if logging is off because it does not evaluate its arguments.

What I worry is those parameters to be evaluated. It seems that (if the docs says what it says) parameters like `toaddr.String()` and `len(packet)` will always be evaluated to its value no matter mlog is enabled or disabled. If it happens to be an expensive operation, then we may be in trouble. For instance, no idea how cheap or expensive this is, but usually string printing like `toaddr.String()` is not a cheap operation...